Dockerfile.gcc

```diff
@@ -78,10 +78,48 @@ ARG KBUILD_BUILD_TIMESTAMP
 ARG KERNEL_CONFIG
 ARG KERNEL_BIN
 ARG LOCALVERSION
+
 # ARCH and CROSS_COMPILE are inherited from builder-${TARGETARCH}-${BUILDARCH}
 # ARCH is always set to the target arch
 # CROSS_COMPILE is set to empty string for native builds
-COPY . /kernel-src
+RUN mkdir /kernel-src
+COPY COPYING /kernel-src/COPYING
+COPY CREDITS /kernel-src/CREDITS
+COPY Documentation /kernel-src/Documentation
+COPY Kbuild /kernel-src/Kbuild
+COPY Kconfig /kernel-src/Kconfig
+COPY LICENSES /kernel-src/LICENSES
+COPY MAINTAINERS /kernel-src/MAINTAINERS
+COPY Makefile /kernel-src/Makefile
+COPY README /kernel-src/README
+COPY arch /kernel-src/arch
+COPY block /kernel-src/block
+COPY certs /kernel-src/certs
+COPY crypto /kernel-src/crypto
+COPY drivers /kernel-src/drivers
+COPY fs /kernel-src/fs
+COPY include /kernel-src/include
+COPY init /kernel-src/init
+COPY io_uring /kernel-src/io_uring
+COPY ipc /kernel-src/ipc
+COPY kernel /kernel-src/kernel
+COPY lib /kernel-src/lib
+COPY mm /kernel-src/mm
+COPY net /kernel-src/net
+COPY rust /kernel-src/rust
+COPY samples /kernel-src/samples
+COPY scripts /kernel-src/scripts
+COPY security /kernel-src/security
+COPY sound /kernel-src/sound
+COPY tools /kernel-src/tools
+COPY usr /kernel-src/usr
+COPY virt /kernel-src/virt
+# We have to do this as --exclude is not supported by dockerfile-add-scanner
+# and we cannot split this into several Dockerfiles as the docker buildkit
+# cannot reference local images
+# To generate this, I used:
+# ls -1 | perl -ne 'chomp; print "COPY $_ /kernel-src/$_\n";'
```

Comment:
@rucoder , @christoph-zededa , is really `ls -1` enough? Because it's going to list everything, including Dockerfiles, Makefile.eve, etc.... do you think we should include a proper script to list all required files, exclude the ones we want to exclude and generate the COPY lines?

Reply:
I filtered it manually, so `ls -1` is not enough.
I am not sure if it makes sense to create a proper script as the decision whether a file/dir has to copied or not is not so easy to make.
Sure we could have a list of ignored files/dirs, but then instead of editing the output of `ls -1`, one has to edit the list; i.e. I don't see much gain from such a script.

Reply:
ok, then could you please update the comment `To generate this, I used:` to clarify a bit more, for instance (suggestion), `The list of files can be generated by a command like the following:`, my point is to clarify that only this command might not be enough......

Reply:
I added some text to clarify it: https://github.com/lf-edge/eve-kernel/compare/e621caa498159acd66099403b0e3c81a3763fa62..a56a85e69d6dde9cdd3471614d3fc5b3f0ef0b1e

Reply:
ok, it's better now. I'm not against these bunch of COPY commands but I'm wondering something..... for nvidia kernel, for example, we have the sources inside another folder, not in the root of the repo... what do you think if we do the same here? You can move the needed files to a kernel-src folder, for example, and then we can just copy this folder with a single COPY command....

Reply:
> what do you think if we do the same here?

I think the problem is that you then cannot rebase anymore on `github.com/torvalds/linux` or do a cherry-pick etc...

Reply:
I thought about some alternatives:
* fix dockerfile-add-scanner
* run dockerfile-add-scanner within this docker container and not separately
* create another repository only with `Makefile.eve`, `Dockerfile.*` which has this repo as a submodule, `repo.sh`, `git clone`, or whatever ...
* put `Makefile.eve`, `Dockerfile.*` into eve's repository under `pkg/kernel`

But I haven't found a definitive good answer.